src/lib/comms/include/sol-http-server.h

```diff
@@ -94,10 +95,35 @@ struct sol_http_progressive_response;
  * @param port The port where the server will bind.
  *
  * @return a handle to the server on success, otherwise @c NULL is returned.
+ *
+ * @see sol_http_server_new()
  */
 struct sol_http_server *sol_http_server_new(uint16_t port);
 
 /**
+ * @brief Creates a HTTPS server, binding on all interfaces
+ * in the specified @a port.
+ *
+ * With the returned handle it's possible to register paths using
+ * @c sol_http_server_register_handler()
+ * and dirs with @c sol_http_server_add_dir().
+ *
+ * @note Only one instance of sol_http_server is possible per port.
+ * Try to run a second instance in the
+ * same port will result in fail.
+ *
+ * @param port The port where the server will bind.
+ * @param cert The certificate that should be used to crypto the data.
+ * @param key The key of @a cert.
+ *
+ * @return a handle to the server on success, otherwise @c NULL is returned.
+ *
+ * @see sol_http_server_new()
+ */
+struct sol_http_server *sol_http_server_secure_new(uint16_t port,
```

Comment:
On 21-04-2016 11:00, Flávio Vinicius Alvares Ceolin wrote:
> In src/lib/comms/include/sol-http-server.h
> <https://github.com/solettaproject/soletta/pull/1911#discussion_r60584726>:
>
>> + * @c sol_http_server_register_handler()
>> + * and dirs with @c sol_http_server_add_dir().
>> + *
>> + * @note Only one instance of sol_http_server is possible per port.
>> + * Try to run a second instance in the
>> + * same port will result in fail.
>> + *
>> + * @param port The port where the server will bind.
>> + * @param cert The certificate that should be used to crypto the data.
>> + * @param key The key of @a cert.
>> + *
>> + * @return a handle to the server on success, otherwise @c NULL is returned.
>> + *
>> + * @see sol_http_server_new()
>> + */
>> +struct sol_http_server *sol_http_server_secure_new(uint16_t port,
>
> i was trying not introduce another type, but ... if you want guys I can
> proceed

@barbieri wants guys?

>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/solettaproject/soletta/pull/1911/files/74ac3dc1733b8c868d7c68be0bcbe014d54490e3..4a631cb8b713bba1a8ec3f76a2941070a0f28e4e#r60584726>
>